graphql/subscription/map_async_iterator.py

```diff
@@ -19,35 +21,62 @@ def __init__(self, iterable: AsyncIterable, callback: Callable,
         self.iterator = iterable.__aiter__()
         self.callback = callback
         self.reject_callback = reject_callback
-        self.stop = False
+        self._close_event = Event()
+
+    @property
+    def closed(self) -> bool:
+        return self._close_event.is_set()
+
+    @closed.setter
+    def closed(self, value: bool) -> None:
+        if value:
+            self._close_event.set()
+        else:
+            self._close_event.clear()
 
     def __aiter__(self):
         return self
 
     async def __anext__(self):
-        if self.stop:
+        if self.closed:
```

Comment:
Can't we just unconditionally raise StopAsyncIteration here?

Reply:
Initially that's what I was doing, but the extra logic was needed to keep this test passing: https://github.com/graphql-python/graphql-core-next/blob/170b227e6cbf67d2cc35555c7ecabbdfcf82d7ac/tests/subscription/test_map_async_iterator.py#L84

Reply:
Ah, yes, the finally block should still run.